src/main/java/com/github/shyiko/mysql/binlog/event/deserialization/EventHeaderV4Deserializer.java

```diff
@@ -38,11 +38,15 @@ public EventHeaderV4 deserialize(ByteArrayInputStream inputStream) throws IOExce
         return header;
     }
 
+    /**
+     * Parses the event type based on the ordinal.
+     * If an invalid or proprietary ordinal is passed, an unknown event is returned.
+     */
     private static EventType getEventType(int ordinal) throws IOException {
         EventType eventType = EventType.forId(ordinal);
 
         if (eventType == null) {
-            throw new IOException("Unknown event type " + ordinal);
+            return EventType.forId(0);
```

Comment:
I think you can just write this as `return EventType.UNKNOWN;`

Reply:
Good call, changed